packages/cashier-v2/src/components/NewsTicker/NewsTicker.tsx

```diff
@@ -0,0 +1,47 @@
+import React, { useState } from 'react';
+import clsx from 'clsx';
+import NewsTickerChildren from './NewsTickerChildren';
+import styles from './NewsTicker.module.scss';
+
+type TNewsTicker = {
+    children: React.ReactNode;
+    className?: string;
+    speed: number;
+};
+
+const NewsTicker: React.FC<TNewsTicker> = ({ children, className, speed }) => {
+    const [elementWidth, setElementWidth] = useState(-1);
+    const [isExceedingParent, setIsExceedingParent] = useState(false);
+
+    const onRefChange = (ref: HTMLDivElement) => {
+        if (ref) {
+            setIsExceedingParent(ref.scrollWidth > ref.clientWidth);
+
+            if (elementWidth === -1) {
+                setElementWidth(ref.scrollWidth);
+            }
+        }
+    };
+
+    const animationDuration = elementWidth / speed; // time = distance / speed
+
+    return (
+        <div className={clsx(styles.container, className)} ref={onRefChange}>
+            <NewsTickerChildren
+                animationDuration={animationDuration}
+                isExceedingParent={isExceedingParent}
+                reactChildren={children}
+            />
+            {isExceedingParent && (
+                <NewsTickerChildren
+                    animationDuration={animationDuration}
+                    isExceedingParent={isExceedingParent}
+                    isSecondContainer
```

Comment:
as a rule of thumb, component should not be aware of its sorroundings. So just "animationDelay"? 

difference is, that with "animationDelay" parameter, code reader immediately knows the meaning of this parameter, and it will work pretty much anywhere (including the use as thirdContainer, fourthContainer or zeroContainer) and its gonna be immediately readable,

Reply:
similiarly about "isExceedingParent"

ideally, it should be a paraemtere like "enableAnimation" (basically, something based on what the parameter does, and not based on where component exisits)

Reply:
sure @wojciech-deriv 
we can fix it in upcoming PR's
thanks for pointing on this